aidants_connect_web/views/usagers.py

```diff
@@ -38,17 +38,25 @@ def usager_details(request, usager_id):
         django_messages.error(request, "Cet usager est introuvable ou inaccessible.")
         return redirect("dashboard")
 
-    active_autorisations = aidant.get_active_autorisations_for_usager(usager_id)
-    inactive_autorisations = aidant.get_inactive_autorisations_for_usager(usager_id)
+    active_mandats = (
+        Mandat.objects.prefetch_related("autorisations")
```

Comment:
👍 pour l'optimisation, mais je me demande si dans ce cas ça n'est pas plutôt [`select_related`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#select-related) qu'il conviendrait d'utiliser — est-ce que tu es sûr de toi quant à ce choix ? 😅

Reply:
tu m'as mis le doute, mais il semblerait que 
- select_related soit pour les "single-valued relationships" (foreign key and one-to-one)
- prefetch_related soit pour les "many-to-many and many-to-one objects"
cf [ce paragraphe](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#prefetch-related)

En fait l'usage que je veux en faire, c'est de pouvoir loop ensuite sur les autorisations sans avoir à aller les chercher une par une. Mais je viens de tester dans `shell_plus --print-sql` et je ne vois pas de différence avec et sans le `prefetch_related`... 🤔

Reply:
Justement, dans ce cas, on est sur une FK non ?
Ce qui semblerait indiquer que la bonne optim est `select_related`.
Auquel cas, tu _devrais_ voir des `JOIN` en plus avec `--print-sql` :)

Reply:
Vu sur Slack, c'est bien une FK, mais "del'autre côté", donc l'optim à utiliser est bien `prefetch_related`, my bad.

Ceci dit, au delà de l'aspect didactique, c'est de toutes façons très prématuré en termes de besoins réels 😂